tests/nonbonded/test_nonbonded_interaction_group.py

```diff
@@ -79,26 +67,14 @@ def test_nonbonded_interaction_group_correctness(
     conf = example_conf[:num_atoms]
     params = example_nonbonded_potential.params[:num_atoms, :]
 
-    lambda_plane_idxs = rng.integers(-2, 3, size=(num_atoms,), dtype=np.int32)
-    lambda_offset_idxs = rng.integers(-2, 3, size=(num_atoms,), dtype=np.int32)
-
     ligand_idxs = rng.choice(num_atoms, size=(num_atoms_ligand,), replace=False).astype(np.int32)
 
-    potential = generic.NonbondedInteractionGroup(
-        ligand_idxs,
-        lambda_plane_idxs,
-        lambda_offset_idxs,
-        beta,
-        cutoff,
-    )
-
-    lambda_vals = [0.0, 0.1]
+    potential = generic.NonbondedInteractionGroup(num_atoms, ligand_idxs, beta, cutoff)
 
     GradientTest().compare_forces_gpu_vs_reference(
         conf,
-        params,
+        gen_nonbonded_params_with_4d_offsets(rng, params, -2 * cutoff, 2 * cutoff),
```

Comment:
what's the rationale for going beyond the cutoff?

Reply:
This is just for consistency with the existing tests, which tested lambda plane indices in {-2, 1, 0, 1, 2} (e.g. [here](https://github.com/proteneer/timemachine/pull/893/files#diff-bb6ec9fd4f933d4ba34a7034486e1d1d3be687475b700a38143fa656c04f9a89L82-L83)). It also seems useful to test nontrivial cases where some pairs are fully decoupled

Reply:
makes sense!

Reply:
After some more thought, I addressed this in 2914ed4. I think you are right that there's no benefit to testing `w` in `[-2 * cutoff, 2 * cutoff]`, since even with `[-cutoff, cutoff]` there will be fully decoupled pairs. Also fixed the generation of negative pair `delta_w`s in the test for `NonbondedPrecomputed`.